packages/hardhat-core/src/internal/sentry/anonymizer.ts

```diff
@@ -226,4 +236,108 @@ export class Anonymizer {
 
     return result;
   }
+
+  private _anonymizeMnemonic(errorMessage: string): string {
+    const mnemonicWordlist = ([] as string[]).concat(
+      ...[
+        require("ethereum-cryptography/bip39/wordlists/czech"),
+        require("ethereum-cryptography/bip39/wordlists/english"),
+        require("ethereum-cryptography/bip39/wordlists/french"),
+        require("ethereum-cryptography/bip39/wordlists/italian"),
+        require("ethereum-cryptography/bip39/wordlists/japanese"),
+        require("ethereum-cryptography/bip39/wordlists/korean"),
+        require("ethereum-cryptography/bip39/wordlists/simplified-chinese"),
+        require("ethereum-cryptography/bip39/wordlists/spanish"),
+        require("ethereum-cryptography/bip39/wordlists/traditional-chinese"),
+      ].map((wordlistModule) => wordlistModule.wordlist)
+    );
+    const matches = getAllWordMatches(errorMessage);
+    // If there are enough consecutive words, there's a good chance of there being a mnemonic phrase
+    if (matches.length >= MNEMONIC_PHRASE_LENGTH_THRESHOLD) {
+      let anonymizedMessage = "";
+      // Determine all mnemonic phrase maximal fragments.
+      // We check sequences of n consecutive words just in case there is a typo
+      for (let wordIndex = 0; wordIndex < matches.length; wordIndex++) {
+        const maximalPhrase = getMaximalMnemonicPhrase(
+          matches,
+          errorMessage,
+          wordIndex,
+          mnemonicWordlist
+        );
+
+        if (maximalPhrase.length >= MINIMUM_AMOUNT_OF_WORDS_TO_ANONYMIZE) {
+          const lastAnonymizedWord = maximalPhrase[maximalPhrase.length - 1];
+          const nextWordIndex =
+            wordIndex + maximalPhrase.length < matches.length
+              ? matches[wordIndex + maximalPhrase.length].index
+              : errorMessage.length;
+          const sliceUntilNextWord = errorMessage.slice(
+            lastAnonymizedWord.index + lastAnonymizedWord.word.length,
+            nextWordIndex
+          );
+          anonymizedMessage += `${ANONYMIZED_MNEMONIC}${sliceUntilNextWord}`;
+          wordIndex += maximalPhrase.length - 1;
```

Comment:
The `-1` is here to cancel the `++` of the for loop, right? That makes me think that this shoul be a `while` instead.

Reply:
That's right. I was actually thinking the same.